appkeeper/datadog_checks/appkeeper/appkeeper.py

```diff
@@ -0,0 +1,108 @@
+import json
+import requests
+from datetime import datetime, timedelta, timezone
+from dateutil.parser import parse
+from datadog_checks.base.errors import CheckException
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+API_URL = 'https://api.appkeeper.sios.com/v2/'
+AUTH_API_URL = API_URL + 'authorize'
+EVENT_API_URL = API_URL + '{}/events'
+INSTANCES_API_URL = API_URL + '{}/instances'
+
+class AppKeeperCheck(AgentCheck):
+    def check(self, instance, now=None):
+        account = instance.get('account')
+        integrationToken = instance.get('integrationToken')
+
+        if not account or not integrationToken:
+            raise ConfigurationError('Configuration error, please fix conf.yaml')
+
+        token = self.get_token(integrationToken)
+
+        # events
+        events = self.get_events(account, token)
+        if now == None:
+            now = datetime.now(timezone.utc)
+        recover_count = get_recover_count(events, now)
+
+        # instances
+        instances = self.get_instances(account, token)
+        number_of_all_instances = len(instances)
+        number_of_monitored_instances = get_monitoring_instances_number(instances)
+
+        # send data
+        self.gauge('appkeeper.api_recover_count', recover_count)
+        self.gauge('appkeeper.monitored_instances', number_of_monitored_instances)
+        self.gauge('appkeeper.all_instances', number_of_all_instances)
+
+    def get_token(self, integrationToken):
+        return call_auth_api(integrationToken)
+
+    def get_events(self, account, token):
+        return call_events_api(account, token)
+
+    def get_instances(self, account, token):
+        return call_instances_api(account, token)
+
+def call_auth_api(integrationToken):
+    # Get API key
+    try:
+        response = requests.get(AUTH_API_URL + '?token=' + integrationToken)
```

Comment:
```suggestion
        response = self.http.get(AUTH_API_URL + '?token=' + integrationToken)
```
The base AgentCheck class has an HTTP requests wrapper. It supports additional configuration out-of-the-box, like proxy, tls, and additional support.
https://datadoghq.dev/integrations-core/base/http/